Core/include/Acts/EventData/TrackStateType.hpp

```diff
@@ -39,27 +39,31 @@ class TrackStateType {
   using raw_type = std::uint64_t;
   static constexpr std::size_t kRawBits =
       std::numeric_limits<std::make_unsigned<raw_type>::type>::digits;
+
+  // Delete default constructor
+  TrackStateType() = delete;
+
   /// Constructor from a reference to the underlying value container
   /// @param raw the value container
-  TrackStateType(raw_type& raw) : m_raw{&raw} {}
+  TrackStateType(raw_type& raw) : m_raw{&raw} { assert(m_raw != nullptr); }
```

Comment:
I still think this is unconventional, since using a reference puts this in the caller explicitly, but it doesn't hurt either.

Reply:
I saw that we put asserts on this pointer in all the member functions and thought I could just lift this to the constructor so it is guaranteed once

Reply:
Again, no strong opinion, but they're not quite the same imo. 

The assert in the constructor is shielding against an invalid argument, which in principle is avoided by the reference. The assert in the methods is shielding against the member pointer becoming null for some reason. 

It's a bit academic here, so anything you choose is fine.